### PR TITLE
Allow filtering RDS instances by tags in the ec2 dynamic inventory script

### PR DESCRIPTION
**Pull request by jchristi**
##### SUMMARY

Follow up to #23989, which enabled fetching of RDS tags as part of the ec2 dynamic inventory script. One unfortunate side effect of RDS support in ec2.py though is that it does not respect the tag filter settings in ec2.ini. This PR adds that support.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

`contrib/inventory/ec2.py`

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel 8517fbf936) last updated 2017/04/21 12:14:39 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/jchristi/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/share/ansible/lib/ansible
  executable location = /usr/local/share/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

### CODE REVIEW

**Comment by jchristi** [comment]
cc @s-hertel

File: contrib/inventory/ec2.py
Comment (s-hertel) [inline contrib/inventory/ec2.py:621]: Thanks for your work on this! It works as expected (although I wish describe_db_instances() could filter by tags as it would be simpler not to filter out the results while retrieving them).

**Review by s-hertel — APPROVED** [review]

**Comment by s-hertel** [comment]
shipit

File: contrib/inventory/ec2.py
Comment (jchristi) [reply contrib/inventory/ec2.py:621]: Agreed. It doesn't appear the RDS API supports that yet :(

**Comment by jchristi** [comment]
Its been a month since this was approved. Can we get this merged? Thanks!

**Comment by s-hertel** [comment]
cc @ryansb @willthames @mikedlr Can I get another set of eyes on this before merging, please?

**Comment by aioue** [comment]
@s-hertel tested. This works as described - inventory output is restricted to RDS instances matching the tags specified in the `.ini`.

**Comment by s-hertel** [comment]
Thanks so much for testing this @aioue!

**Comment by s-hertel** [comment]
rebuild_merge

**Pull request merged**
